File: python/ordenamiento/shellsort/ShellSortTest.py
def ordenarShell(arreglo): 

	arreglo_size = len(arreglo) 
	brecha = int(arreglo_size/2) #división entera
	
	while brecha > 0: 		
		for recorrido in range(brecha,arreglo_size): 			
			buffer = arreglo[recorrido] 
			indice = recorrido 
			while indice >= brecha and arreglo[indice-brecha] > buffer: 
				arreglo[indice] = arreglo[indice-brecha] 
				indice -= brecha 
			arreglo[indice] = buffer 
			
		brecha = int(brecha/2) #división entera

# Función principal 
if __name__ == "__main__":
	datos = [8, 5, 3, 9, 1, 4, 7]
	arreglo_size = len(datos) 
	print ("Arreglo original: " + str(datos)) 
	ordenarShell(datos) 
	print ("Arreglo original: " + str(datos)) 

File: java/ordenamiento/shellsort/ShellSortTest.java
import java.util.Arrays;

public class ShellSortTest {
	
	static void ordenarShell(int arreglo[]) {
		int arreglo_size = arreglo.length;
		int brecha = (arreglo_size/2);
		
		while (brecha > 0) { 				
			for (int recorrido=brecha; recorrido < arreglo_size ; recorrido++) { 
				int buffer = arreglo[recorrido];
				int indice = recorrido ;
				while (indice >= brecha && arreglo[indice-brecha] > buffer) { 
					arreglo[indice] = arreglo[indice-brecha];
					indice -= brecha;
				}
				arreglo[indice] = buffer;
			}
			brecha = brecha/2;
		}
	}

	public static void main(String args[]) {
		int datos[] = new int[]{8, 5, 3, 9, 1, 4, 7};
		System.out.println("Arreglo original: " + Arrays.toString(datos));
		ordenarShell(datos);
		System.out.println("Arreglo ordenado: " + Arrays.toString(datos));
	}
}

